tests/main/searching/task.yaml

```diff
@@ -41,6 +41,11 @@ execute: |
     expected='(?s)Name +Version +Publisher +Notes +Summary *\n(.*?\n)?test-snapd-tools +.*? *\n.*'
     snap find test-snapd- | grep -Pzq "$expected"
 
+    echo "And a very specific query works too"
+    # and returns a single result
+    expected='^Name +Version +Publisher +Notes +Summary\ntest-snapd-python-webserver +.*\n$'
+    snap find python based example webserver test-snapd | grep -Pzq "$expected"
```

Comment:
This fails on ubuntu-core-16 because its version of grep is old and has a bug related to -P and the anchors:
```
grep: + snap find python based example webserver test-snapd
unescaped ^ or $ not supported with -Pz
```

Reply:
Maybe it's safer to dump the output into a file, and then check the two lines one by one with two different grep commands; last, check that `wc -l` returns 2